libstuff/libstuff.cpp

```diff
@@ -1910,6 +1911,31 @@ string SAESEncrypt(const string& buffer, unsigned char* iv, const string& key) {
     return encryptedBuffer;
 }
 
+string SAESEncrypt(char* buffer, const int bufferSize, const string& ivStr, const string& key) {
```

Comment:
Can the existing `string SAESEncrypt(const string& buffer, unsigned char* iv, const string& key)` just call this one?

I.e.,
```
string SAESEncrypt(const string& buffer, unsigned char* iv, const string& key) {
    return SAESEncrypt(buffer.c_str(), buffer.size(), iv, key);
}
```